app/Services/UpdateEmailService.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Services;
+
+use Brewmap\Eloquent\User;
+use Brewmap\Exceptions\User\NewEmailChangingException;
+use Brewmap\Notifications\EmailChangeNotification;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Notification;
+
+class UpdateEmailService
+{
+    /**
+     * Changes the user Email Address for a new one
+     */
+    public function sendNotifyForNewEmail(string $email): void
+    {
+        Notification::route("mail", $email)
+            ->notify(new EmailChangeNotification(auth()->user()->id));
```

Comment:
And think about using `auth()` here. What would happen with this service ran from queued event?